### ISSUE TITLE
Panic: Invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by kody-abe** [aws/amazon-vpc-cni-k8s#1108]
```
====== Installing AWS-CNI ======
====== Starting amazon-k8s-agent ======
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x55cd89da9954]
goroutine 1 [running]:
github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils.(*EC2InstanceMetadataCache).DescribeENI(0xc0001f05a0, 0xc000041720, 0x15, 0x55cd8ab2e9e0, 0xc0001cd7a0, 0x55cd8aae3ba0, 0xc00064bb48, 0x55cd88dbe953, 0x55cd8aaa2b70)
/go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils/awsutils.go:820 +0x334
github.com/aws/amazon-vpc-cni-k8s/ipamd.(*IPAMContext).getENIaddresses(0xc000380000, 0xc000041720, 0x15, 0xc000048070, 0xc000048000, 0xc0001cd7a0, 0x55cd8aae3ba0, 0xc0011845d0, 0xc00064bbd0, 0x55cd88dbe953)
/go/src/github.com/aws/amazon-vpc-cni-k8s/ipamd/ipamd.go:754 +0x53
github.com/aws/amazon-vpc-cni-k8s/ipamd.(*IPAMContext).setupENI(0xc000380000, 0xc000041720, 0x15, 0xc000041720, 0x15, 0xc000042953, 0x11, 0x3, 0xc000589390, 0xc, ...)
/go/src/github.com/aws/amazon-vpc-cni-k8s/ipamd/ipamd.go:709 +0x5c
github.com/aws/amazon-vpc-cni-k8s/ipamd.(*IPAMContext).nodeInit(0xc000380000, 0x0, 0x0)
/go/src/github.com/aws/amazon-vpc-cni-k8s/ipamd/ipamd.go:301 +0x947
github.com/aws/amazon-vpc-cni-k8s/ipamd.New(0x55cd8aae39a0, 0xc0005401c0, 0xc0004108a0, 0x1a, 0x1, 0x0)
/go/src/github.com/aws/amazon-vpc-cni-k8s/ipamd/ipamd.go:242 +0x244
main._main(0x0)
/go/src/github.com/aws/amazon-vpc-cni-k8s/main.go:61 +0x326
main.main()
/go/src/github.com/aws/amazon-vpc-cni-k8s/main.go:38 +0x24
```

**Comment by mogren** [comment]
@kody-abe Thanks for reporting, what version of the CNI is this?

**Comment by wundrian** [comment]
Same error (different stack, being called from `FreeENI`). Our CNI version is at v1.5.7

**Comment by SaranBalaji90** [comment]
This is caused by https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.5.3/pkg/awsutils/awsutils.go#L820

**Comment by SaranBalaji90** [comment]
Fixed with this commit https://github.com/aws/amazon-vpc-cni-k8s/pull/1001

**Comment by SaranBalaji90** [comment]
@kody-abe @wundrian sorry for the delayed response here. Do you still have this instance running? Because my guess is one of the ENI attached to the instance might not be returning "Attachment" information in the response. If you still have the ENI and noticing this, I can reach out to the respective internal teams to verify why we are not seeing the attachment details.

Sample command:
aws ec2 describe-network-interfaces --network-interface-ids eni-07dcc793e76a5328c

**Comment by mogren** [comment]
Closing because it was fixed in #1001.
